dojo_plugin/api/v1/docker.py

```diff
@@ -5,25 +5,42 @@
 import subprocess
 import sys
 import traceback
+import logging
 
 import docker
-from flask import request, current_app
+from flask import abort, request, current_app
 from flask_restx import Namespace, Resource
+from CTFd.exceptions import UserNotFoundException, UserTokenExpiredException
 from CTFd.utils.user import get_current_user, is_admin
 from CTFd.utils.decorators import authed_only
 
 from ...config import HOST_DATA_PATH, INTERNET_FOR_ALL, WINDOWS_VM_ENABLED, SECCOMP, USER_FIREWALL_ALLOWED
 from ...models import Dojos, DojoModules, DojoChallenges
-from ...utils import serialize_user_flag, resolved_tar, random_home_path, module_challenges_visible, user_ipv4
+from ...utils import (
+    lookup_workspace_token, 
+    serialize_user_flag, 
+    resolved_tar, 
+    random_home_path, 
+    module_challenges_visible, 
+    user_ipv4
+)
 from ...utils.dojo import dojo_accessible, get_current_dojo_challenge
 from ...utils.workspace import exec_run
 
+logger = logging.getLogger(__name__)
+
 docker_namespace = Namespace(
     "docker", description="Endpoint to manage docker containers"
 )
 
 
-def start_challenge(user, dojo_challenge, practice):
+def start_challenge(owner, challenged_user, dojo_challenge, practice, partner=None):
```

Comment:
Lets use this interface:
```py
def start_challenge(user, dojo_challenge, practice, *, as_user=None):
    as_user = as_user or user
```

A user starts a challenge, and they might start that challenge "as" another user (by default it's "as" themself).

Basically everything will use `user`, except:
- Mount `/home/hacker` from `as_user`
- Mount `/home/me` from `user`, but only mounted if `user != as_user` else no mount.
- Pass `as_user` to `insert_challenge` instead of `user`.